### PR TITLE
DOC: Improved docs for `jnp.median` and `nanmedian`

### PR DESCRIPTION
**Pull request by rajasekharporeddy**
Part of #21461

### CODE REVIEW

File: jax/_src/numpy/reductions.py [hunk 1]
```diff
@@ -2069,20 +2069,109 @@ def nanpercentile(a: ArrayLike, q: ArrayLike,
...
            out: None = None, overwrite_input: bool = False,
            keepdims: bool = False) -> Array:
+  r"""Return the median of array elements along a given axis.
+
+  JAX implementation of :func:`numpy.median`.
+
+  Args:
+    a: input array.
+    axis: optional, int or sequence of ints, default=None. Axis along which the
+      median to be computed. If None, median is computed along all the axes.
```
Comment (dfm) [inline jax/_src/numpy/reductions.py:2084]:
```suggestion
      median to be computed. If None, median is computed for the flattened array.
```

File: jax/_src/numpy/reductions.py [hunk 2]
```diff
@@ -2069,20 +2069,109 @@ def nanpercentile(a: ArrayLike, q: ArrayLike,
...
               out: None = None, overwrite_input: bool = False,
               keepdims: bool = False) -> Array:
+  r"""Return the median of array elements along a given axis, ignoring NaNs.
+
+  JAX implementation of :func:`numpy.nanmedian`.
+
+  Args:
+    a: input array.
+    axis: optional, int or sequence of ints, default=None. Axis along which the
+      median to be computed. If None, median is computed along all the axes.
```
Comment (dfm) [inline jax/_src/numpy/reductions.py:2135]:
```suggestion
      median to be computed. If None, median is computed for the flattened array.
```

File: jax/_src/numpy/reductions.py [hunk 3]
```diff
@@ -2069,20 +2069,109 @@ def nanpercentile(a: ArrayLike, q: ArrayLike,
...
+  Returns:
+    An array of the median along the given axis.
+
+  See also:
+    - :func:`jax.numpy.mean`: Compute the mean of array elements over a given axis.
+    - :func:`jax.numpy.max`: Compute the maximum of array elements over given axis.
+    - :func:`jax.numpy.min`: Compute the minimum of array elements over given axis.
+
+  Examples:
+    By default, the median is computed along all the axes.
```
Comment (dfm) [inline jax/_src/numpy/reductions.py:2099]:
```suggestion
    By default, the median is computed for the flattened array.
```

File: jax/_src/numpy/reductions.py [hunk 4]
```diff
@@ -2069,20 +2069,109 @@ def nanpercentile(a: ArrayLike, q: ArrayLike,
...
+  See also:
+    - :func:`jax.numpy.nanmean`: Compute the mean of array elements over a given
+      axis, ignoring NaNs.
+    - :func:`jax.numpy.nanmax`: Compute the maximum of array elements over given
+      axis, ignoring NaNs.
+    - :func:`jax.numpy.nanmin`: Compute the minimum of array elements over given
+      axis, ignoring NaNs.
+
+  Examples:
+    By default, the median is computed along all the axes.
```
Comment (dfm) [inline jax/_src/numpy/reductions.py:2154]:
```suggestion
    By default, the median is computed for the flattened array.
```

**Review by dfm — CHANGES_REQUESTED** [review]
Thanks! I added some comments about the default behavior, but it otherwise looks good to me. Can you rebase onto `main` to fix the docs build error too?

**Comment by rajasekharporeddy** [comment]
Modified the docstrings as suggested and rebased onto main.

File: jax/_src/numpy/reductions.py (hunk 4 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Done

File: jax/_src/numpy/reductions.py (hunk 3 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Done

File: jax/_src/numpy/reductions.py (hunk 2 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Done

**Review by dfm — APPROVED** [review]
Thanks!

**Pull request merged**